GEECS-PythonAPI/geecs_python_api/controls/devices/geecs_device.py

```diff
@@ -121,10 +121,20 @@ def init_resources(self):
                     if self.dev_tcp is None:
                         try:
                             self.dev_tcp = TcpSubscriber(owner=self)
+                            ###try to connect a TCP subscriber to see if device is on and accessible
+                            is_device_on = self.dev_tcp.connect()
                         except Exception:
-                            api_error.error('Failed creating TCP subscriber', 'GeecsDevice class, method "__init__"')
+                            api_error.error('Failed to connect a TCP subscriber', 'GeecsDevice class, method "__init__"')
                 else:
                     api_error.warning(f'Device "{self.__dev_name}" not found', 'GeecsDevice class, method "__init__"')
+                
+                if not is_device_on:
```

Comment:
`is_device_on` may not be defined, if `is_device_on = self.dev_tcp.connect()` raises an Exception.